src/Histogram.ts

```diff
@@ -260,11 +266,12 @@ export default class Histogram {
 
     const bins = new Uint32Array(numBins).fill(0);
 
-    const binSize = (max - min) / numBins === 0 ? 1 : (max - min) / numBins;
+    // Subtract 1 so the last bin includes the max value.
+    const binSize = max - min === 0 ? 1 : (max - min) / (numBins - 1);
```

Comment:
You can see this at work in the unit tests, but this makes it so the last bin has the `max` as the lower bin boundary and lets me remove some special-case logic. `vole-app` is already doing this!

Reply:
I'll write a more practical case: random intensities from 0-255.  We want 8 bins.  The bin bounds should be
```
1 | 0-31
2 | 32-63
3 | 64-95
4 | 96-127
5 | 128-159
6 | 160-191
7 | 192-223
8 | 224-255
```
For this type of data, I think you could write the min of bin 2 as 32-0.5 and the max of bin 1 as 31+0.5

Reply:
One more thing I thought of is that you could pass the dtype in to the Histogram if you had a reason to distinguish between integer values and floating point for the bin shapes.

Reply:
Okay, added a big code comment and refactored the code to handle the float and integer cases!

Reply:
Not meaning to be difficult, but is all the code around calculating the number of bins needed?  The method is
`private static calculateHistogram(arr: TypedArray<NumberType>, numBins = 1)`
and only called once, 
`// build up the histogram
    const hinfo = Histogram.calculateHistogram(data, NBINS);`
and NBINS is a constant 256.
I think to code is correct, but overbuilt for the use case.  If calculateHistogram is intended for wider use in a later update, then it makes more sense.

Reply:
I see what you mean about `calculateHistogram()` only ever being called with 256 bins. It's possible in the future we could have a differently-sized Histogram, but for now I *am* using a 4-wide histogram in some of my unit tests to match the code comment example since it's easier to reason about.

I'm not sure `math.floor` and `math.ceil` will work in the float case? For example, if our data range is a very small float range, (ex. `[0.45, 0.46]`), the Histogram would be from the range [0, 1], and most of the range would be wasted.